code/modules/clothing/suits/cardborg_suit.dm

```diff
@@ -0,0 +1,231 @@
+	item_state = "cardborg_xeno"
+	species_disguise = "High-tech alien-hunting robot"
+
+/obj/item/clothing/suit/cardborg/deathbot
+	name = "black cardborg suit"
+	desc = "A full-body suit made out of ordinary cardboard boxes with various holes cut into them. This one has been spray-painted black."
+	icon_state = "cardborg_deathbot"
+	item_state = "cardborg_deathbot"
+	species_disguise = "High-tech killer robot"
+
+/*
+ * Disguise code.
+*/
+/obj/item/clothing/head/cardborg/equipped(mob/living/user, slot)
+	..()
+	if(ishuman(user) && slot == SLOT_HUD_HEAD)
+		var/mob/living/carbon/human/H = user
+		if(!istype(H.wear_suit, /obj/item/clothing/suit/cardborg))
+			return
+		var/obj/item/clothing/suit/cardborg/CB = H.wear_suit
+		CB.apply_borg_disguise(user, src)
+
+/obj/item/clothing/head/cardborg/dropped(mob/living/user)
+	..()
+	user.remove_alt_appearance("selected_borg_disguise")
+
+/obj/item/clothing/suit/cardborg/equipped(mob/living/user, slot)
+	..()
+	if(ishuman(user) && slot == SLOT_HUD_OUTER_SUIT)
+		var/mob/living/carbon/human/H = user
+		if(!istype(H.head, /obj/item/clothing/head/cardborg))
+			return
+		var/obj/item/clothing/head/cardborg/head = H.head
+		src.apply_borg_disguise(user, head)
+
+/obj/item/clothing/suit/cardborg/dropped(mob/living/user)	
+	..()
+	user.remove_alt_appearance("selected_borg_disguise")
+
+/obj/item/clothing/suit/cardborg/proc/apply_borg_disguise(mob/living/carbon/human/H, obj/item/clothing/head/cardborg/borghead)
+	if(!istype(H) || !istype(borghead))
+		return
+	if(species_disguise != borghead.species_disguise)	// Ensure the head and body are the same colour.
+		to_chat(usr, "<span class='warning'>The colours of the cardborg helmet and suit do not match, the disguise is not convincing enough to work!</span>")
+		return
+	var/selected_borg_disguise = pick(borghead.available_disguises)
+	var/selected_borg_eyes
+	var/list/borgs_with_shared_eyes = list("Cricket", "Standard")	// Some borg eyes are used by multiple different sprites. These cases require special handling.
+	var/list/skin_to_truncate = splittext(selected_borg_disguise, "-")
+	var/truncated_skin = trim(skin_to_truncate[1])	// Slice off the suffix of the skin variant's name so it matches the name of the eyes it is supposed to pair with.
+	if(truncated_skin in borgs_with_shared_eyes)
+		selected_borg_eyes += truncated_skin
+	else
+		selected_borg_eyes = selected_borg_disguise	// Otherwise the eyes are already bespoke for the skin so we just match them.
+	var/image/I = image(icon = 'icons/mob/robots.dmi' , icon_state = selected_borg_disguise, loc = H)	// Now you're a robot!
+	I.override = 1
+	I.overlays += image(icon = 'icons/mob/robots.dmi', icon_state = "eyes-[selected_borg_eyes]")	// Gotta look realistic! Check to see if the borg type has eyes - if yes, apply them.
+	H.add_alt_appearance("selected_borg_disguise", I, GLOB.silicon_mob_list+H)	// You look like a robot to robots (including yourself because you're totally a robot)!
```

Comment:
```suggestion
	H.add_alt_appearance("selected_borg_disguise", I, GLOB.silicon_mob_list + H)	// You look like a robot to robots (including yourself because you're totally a robot)!
```